src/autogluon/cloud/cluster/cluster_config_generator.py

```diff
@@ -0,0 +1,19 @@
+from abc import ABC, abstractmethod
+from typing import Any, Dict
+
+
+class ClusterConfigGenerator(ABC):
+    @abstractmethod
+    @staticmethod
+    def get_default_config() -> Dict[str, Any]:
+        """
+        Get default config of the cluster
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def merge_config(new_config) -> Dict[str, Any]:
```

Comment:
[minor] better to rename to `update_config` unless there is a strong reason to use `merge`